src/test/java/com/amihaiemil/docker/mock/Condition.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2018, Mihai Emil Andronache
+ * All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1)Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2)Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3)Neither the name of docker-java-api nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.amihaiemil.docker.mock;
+
+import java.util.function.Predicate;
+import org.apache.http.HttpRequest;
+
+/**
+ * Condition that an {@link HttpRequest} must satisfy.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 0.0.1
+ */
+public final class Condition {
+    /**
+     * The failure message.
+     */
+    private final String msg;
+    /**
+     * The test that the http request must satisfy.
+     */
+    private final Predicate<HttpRequest> test;
+
+    /**
+     * Ctor.
+     * 
+     * @param msg The failure message.
+     * @param test The test.
+     */
+    public Condition(final String msg, final Predicate<HttpRequest> test) {
+        this.msg = msg;
+        this.test = test;
+    }
+
+    /**
+     * The failure message.
+     * 
+     * @return The failure message.
+     */
+    public String msg() {
+        return this.msg;
+    }
+
+    /**
+     * The test for the http request.
+     * 
+     * @return The test for the http request.
+     */
+    public Predicate<HttpRequest> test() {
+        return this.test;
+    }
```

Comment:
@llorllale This method could be changed a little, like this: 

```java
public void test(final HttpRequest request) {
    if(!this.predicate.test(request)){
        Assert.fail(this.msg);
    };
}
```

Then you could also get rid of the ``msg()`` accessor and also refactor method ``check`` from AssertRequest, it could simply be:

```java
private void check(final HttpRequest request) {
    this.conditions.forEach(cond -> {cond.test(request)});
}
```

I know you said you didn't want a lot of design into this support class, but it's too good not to do it :D 
You can also leave a puzzle for it and I'll do it later, if you spent all the time already.

Reply:
@amihaiemil done